components/builder-api-client/src/builder.rs

```diff
@@ -33,14 +36,26 @@ use std::{fs::{self,
                  PathBuf},
           string::ToString};
 use tee::TeeReader;
-use url::{percent_encoding::{percent_encode,
-                             PATH_SEGMENT_ENCODE_SET},
-          Url};
+use url::Url;
 
 const X_FILENAME: &str = "x-filename";
 
 const DEFAULT_API_PATH: &str = "/v1";
 
+// The characters in this set are copied from
+// https://docs.rs/percent-encoding/1.0.1/percent_encoding/struct.SIMPLE_ENCODE_SET.html
+const PATH_SEGMENT_ENCODE_SET: &AsciiSet = &CONTROLS.add(b' ')
```

Comment:
Forgive my lack of knowledge on this - Why are we defining this ourselves - these all seem to already exist in `percent_encoding::CONTROLS`?

Reply:
No worries. I am lacking knowledge here as well and was surprised we had to manually define the encode set so if there is a better way to do this it would be great.

We were very behind in versions. So far in fact that it has now become two distinct crates `url` and `percent-encoding`.

To me it does not look like the `CONTROLS` variable has those characters. The docs point to this [description](https://url.spec.whatwg.org/#c0-control-percent-encode-set). I believe what we want is described just below that and is the [path percent encode set](https://url.spec.whatwg.org/#path-percent-encode-set).

What version has the `PATH_SEGMENT` variable? That sounds like what I want but I could not find it in the most recent version.

Reply:
Well real talk I could be sending you on a wild goose chase so don't spend too much time on it :joy:. I'm not certain which version, I just found a this change in percent-encoding that looks like it might be the source of `PATH_SEGMENT_ENCODE_SET` being removed https://github.com/servo/rust-url/pull/519/files

Reply:
It looks like what you have is correct, based on my reading of the links you posted, but we might not need '%' and '/'... seems like those are in another set.

That being said, this seems like the kind of very low-level thing that you'd expect a library to handle for you, so it feels like there's something I'm still missing about why we would need this.

:thinking:

Reply:
Whoops, I have the wrong lik in the code comment. It was supposed to be pointing [here](https://docs.rs/percent-encoding/1.0.1/percent_encoding/struct.PATH_SEGMENT_ENCODE_SET.html). That is the documentation for the old constant which includes '%' and '/' and I was trying to preserve the old behavior. Happy to remove them if we decide they are not needed.

I agree it is very odd we have to do this by hand. Especially considering old versions of the library defined the necessary encoding set. However, the [last paragraph](https://docs.rs/percent-encoding/2.1.0/percent_encoding/?search=) of the crates documentation states, "This crate deliberately does not provide many different sets".

Reply:
> However, the [last paragraph](https://docs.rs/percent-encoding/2.1.0/percent_encoding/) of the crates documentation states, "This crate deliberately does not provide many different sets".

Well, that seems pretty straightforward. Guess we get to define our own set. :man_shrugging: 

And it looks like this _is_ replicating the old behavior, so that all looks good :+1: